File: augur-engine/src/collect/trade_sentiment.py
"""Trade system sentiment reader."""

from __future__ import annotations

import json
import os
from datetime import datetime, timezone

from ..config.types import Signal


async def collect_trade_sentiment(path: str | None = None) -> Signal:
    file_path = path or os.environ.get("TRADE_SENTIMENT_PATH", "/tmp/sentiment.json")

    with open(file_path) as f:
        data = json.load(f)

    return Signal(
        source="trade",
        fetched_at=datetime.now(timezone.utc).isoformat(),
        content=data,
    )

File: augur-engine/src/collect/trade-sentiment.ts
import { readFile } from 'node:fs/promises';
import type { Signal } from '../config/types.js';

interface SentimentData {
  sector: string;
  direction: string;
  confidence: number;
  horizon: string;
  rationale_signals: string[];
  generated_at: string;
}

/** Read trade.sh sentiment output from a JSON file */
export async function collectTradeSentiment(path?: string): Promise<Signal> {
  const filePath = path ?? process.env['TRADE_SENTIMENT_PATH'] ?? '/tmp/sentiment.json';

  const raw = await readFile(filePath, 'utf-8');
  const data = JSON.parse(raw) as SentimentData | SentimentData[];

  return {
    source: 'trade',
    fetchedAt: new Date().toISOString(),
    content: data,
  };
}
